src/hardwaresecuritymodules/azext_hardwaresecuritymodules/generated/custom.py

```diff
@@ -0,0 +1,65 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+
+import json
+
+
+def hardwaresecuritymodules_dedicated_hsm_list(cmd, client,
+                                               resource_group_name=None,
+                                               top=None):
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name=resource_group_name,
+                                             top=top)
+    return client.list_by_subscription(top=top)
```

Comment:
@arrownj please help review the usage of `top`

Reply:
@arrownj